blockchain/upgrade.go

```diff
@@ -3252,7 +3313,8 @@ func separateUtxoDatabase(ctx context.Context, db database.DB, utxoDb database.D
 			resumeOffset++
 
 			// Create the new entry in the V3 bucket.
-			err = v3UtxoSetBucketNewDb.Put(cursor.Key(), cursor.Value())
+			newKey := append(utxoPrefixUtxoSetV3, cursor.Key()...)
+			err = tx.Put(newKey, cursor.Value())
```

Comment:
This is a great point.  Per the append docs:

> If it has sufficient capacity, the destination is resliced to accommodate the new elements. If it does not, a new underlying array will be allocated.

In the existing cases, a new array _was_ getting allocated, so no issues were occurring, but this is not really a safe assumption and could easily lead to bugs if things changed.

I added a `prefixedKey` helper function that always creates a new slice and updated all relevant cases to use that instead of `append`.